CMakeLists.txt

```diff
@@ -47,6 +49,30 @@ set(H3_SOVERSION 1)
 
 project(h3 VERSION ${H3_VERSION})
 
+set(H3_COMPILE_FLAGS "")
+set(H3_LINK_FLAGS "")
+if(NOT WIN32)
+    # Compiler options are set only on non-Windows, since these options
+    # are not correct for MSVC.
+    list(APPEND H3_COMPILE_FLAGS -Wall)
+
+    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+        list(APPEND H3_COMPILE_FLAGS
+                -g -gdwarf-2 -g3 -O0 -fno-inline -fno-eliminate-unused-debug-types)
+        if(ENABLE_COVERAGE)
+            list(APPEND H3_COMPILE_FLAGS --coverage)
+            # --coverage is not passed to the linker, so this option is needed
+            # to fully enable coverage.
+            list(APPEND H3_LINK_FLAGS -fprofile-arcs)
```

Comment:
I think `--coverage` would work here too (at least for GCC and clang). But you are right for adding the linker flags in general.